src/core/defs.jl

```diff
@@ -547,9 +547,82 @@ the list unless one of the keywords, `first`, `last`, `before`, `after`.
 """
 function replace_comp!(md::ModelDef, comp_id::ComponentId, comp_name::Symbol=comp_id.comp_name;
                            first::VoidInt=nothing, last::VoidInt=nothing,
-                           before::VoidSymbol=nothing, after::VoidSymbol=nothing)
-    delete!(md, comp_name)
+                           before::VoidSymbol=nothing, after::VoidSymbol=nothing,
+                           reconnect::Bool=true)
+
+    if ! haskey(md.comp_defs, comp_name)
+        error("Cannot replace '$comp_name'; component not found in model.")
+    end
+
+    # Get original position if new before or after not specified
+    # if before == nothing && after == nothing
+    #     comps = collect(keys(md.comp_defs))
+    #     next_idx = findfirst(comps, comp_name) + 1
+    #     before = comps[next_idx]
+    # end 
+
+    # # Get original first and last if new run period not specified
+    # first = first == nothing ? md.comp_defs[comp_name].first : first
+    # last = last == nothing? md.comp_defs[comp_name].last : last
+
+    if reconnect
+        # Assert that new component definition has same parameters and variables needed for the connections
+
+        old_comp = md.comp_defs[comp_name]
+        new_comp = compdef(comp_id)
+
+        # datum_map = (k, v) -> (k, v.datatype, v.dimensions)  # Want to compare datum names, type, and dimensions as a tuple. Should we compare all values of the DatumDef?
+        datum_map = kv -> (kv[1], kv[2].datatype, kv[2].dimensions)  # Want to compare datum names, type, and dimensions as a tuple. Should we compare all values of the DatumDef?
```

Comment:
Curious why the more complicated `kv` version is used instead of `(k, v)`. Do some values have length != 2?